starter-code/src/App.js

```diff
@@ -1,21 +1,234 @@
+          </div>
+          <div className="media-right">
+            <div className="field has-addons">
+              <div className="control">
+                {/* <input className="input" type="number" defaultValue={this.props.quantity}  onChange={(e) => this.props.addQuantity(this.props.name,e.target.value)}/> */}
+                <input className="input" type="number" value={this.state.qty} onChange={(e) => this.handleChange(e)} />
+              </div>
+              <div className="control">
+                <button className="button is-info" onClick={() =>{this.handleClick()}}>
+                  +
+      </button>
+              </div>
+            </div>
+          </div>
+        </article>
       </div>
-    );
+    )
+  }
+}
+
+
+class FormAdd extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      name: "",
+      calories: 0,
+      img: "",
+      quantity: 1,
+    }
+  }
+
+  handleChange(event) {
+    let { name, value } = event.target;
+    if (name === this.state.calories || name === this.state.quantity) {
+      value = parseInt(value);
+    }
+    this.setState({ [name]: value });
+  }
+  // newFood() {
+  //   this.props.displayForm
+  //   this.setState({ displayForm: false })
+  // }
+
+  handleSubmitButtonForm() {
+    const foodAdd = this.state
+    this.props.handleForm()
+    this.props.handleFormSubmit(foodAdd)
+  }
+
+  render() {
+    return (
+      <div>
+        <form>
+          <label for="food">Name: </label>
+          <input type="text" id="food" name="name" value={this.state.name} onChange={(e) => this.handleChange(e)} />
+          <label for="calories">Number of calories: </label>
+          <input type='number' id="calories" name="calories" value={this.state.calories} onChange={(e) => this.handleChange(e)} />
+          <label for='image'>Image: </label>
+          <input type='text' id="image" name="img" value={this.state.img} onChange={(e) => this.handleChange(e)} />
+          <button onClick={() => this.handleSubmitButtonForm()} >Submit</button>
+          {/* <input type="submit" value="Submit" onClick={()=>this.setState({displayForm: false})} /> */}
+        </form>
+      </div>);
   }
 }
 
-export default App;
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { foods, listFoods: foods, selectedFood: [], displayForm: false, searchText: "", displayButtonForForm: true }
+
+  }
+
+  toggleTitle() {
+    this.setState({ displayForm: !this.state.displayForm })
+    this.setState({ displayButtonForForm: !this.state.displayButtonForForm })
+  }
+
+  // handleChange() {
+
+  // }
+
+  addNewFood(foodAdd) {
+    var arrayFood = this.state.listFoods
+    arrayFood.push(foodAdd)
+    this.setState({ listFoods: arrayFood })
+  }
+
+  searchFilter(food) {
+    return food.name.includes(this.state.searchText)
+  }
+
+  addAppQuantity(name, val) {
+    // var arraySelected = {};
+    var selectedFoodCopy = this.state.selectedFood
```

Comment:
Technically, you are not making a copy here. In order to make a copy of the state you can use the slice method like so:
```js
var selectedFoodCopy = this.state.selectedFood.slice()
```
But bravo for using the best practices of React (always making copy of state and never modifying it directly) 👏